app/forms/user_piv_cac_setup_form.rb

```diff
@@ -0,0 +1,80 @@
+class UserPivCacSetupForm
+  # extend Forwardable
+  include ActiveModel::Model
+
+  attr_accessor :x509_dn_uuid, :x509_dn, :token, :user, :nonce, :error_type
+
+  validates :token, presence: true
+  validates :nonce, presence: true
+  validates :user, presence: true
+
+  def submit
+    success = valid? && valid_token?
+
+    FormResponse.new(success: success && process_valid_submission, errors: {})
+  end
+
+  private
+
+  def process_valid_submission
+    user.x509_dn_uuid = x509_dn_uuid
+    user.save!
+    Event.create(user_id: user.id, event_type: :piv_cac_enabled)
+    true
+  rescue PG::UniqueViolation
+    self.error_type = 'piv_cac.already_associated'
+    false
+  end
```

Comment:
This method is violating the [Command-Query Separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation) principle which, according to Wikipedia: 

> states that every method should either be a command that performs an action, or a query that returns data to the caller, but not both. In other words, Asking a question should not change the answer.[1] More formally, methods should return a value only if they are referentially transparent and hence possess no side effects.

What do you think about only calling `process_valid_submission` if the Form passes validation, which is the pattern we follow in our other Form objects?